docs/content/guides/integrations/writing-a-multi-asset-decorator-integration.mdx

```diff
@@ -0,0 +1,290 @@
+    group_name="replication",
+)
+def my_assets(context: AssetExecutionContext):
+    results = replicate(Path(replication_project_path))
+    for table in results:
+        if table.get("status") == "SUCCESS":
+            yield AssetMaterialization(asset_key=str(table.get("name")), metadata=table)
+```
+
+You may have noticed, however, that there are a couple of limitations to this approach. Most importantly, we have not encapsulated the logic for replicating tables. Meaning, end-users who make use of the `custom_replication_assets` decorator would be responsible for yielding asset materializations themselves. Second, there is no way for the user to customize the attributes of the asset.
+
+For the first limitation, we can resolve them by refactoring the code in the body of our asset function into a `Resource`.
+
+---
+
+## Step 3: Moving our replication logic into a Resource
+
+To accomplish this, we will extend the `ConfigurableResource` object to create our own custom resource. Then, we will define a `run` method that will perform the replication operation.
+
+```python
+from dagster import ConfigurableResource
+from dagster._annotations import public
+
+
+class ReplicationResource(ConfigurableResource):
+    @public
+    def run(
+        self, replication_project: ReplicationProject
+    ) -> Iterator[AssetMaterialization]:
+        results = replicate(Path(replication_project.replication_configuration_yaml))
+        for table in results:
+            if table.get("status") == "SUCCESS":
+                # NOTE: this assumes that the table name is the same as the asset key
+                yield AssetMaterialization(
+                    asset_key=str(table.get("name")), metadata=table
+                )
+```
+
+Now, we can refactor our `custom_replication_assets` instance to use this resource.
+
+```python
+@custom_replication_assets(
+    replication_project=replication_project,
+    name="my_custom_replication_assets",
+    group_name="replication",
+)
+def my_assets(replication_resource: ReplicationProject):
+    replication_resource.run(replication_project)
+```
+
+---
+
+## Step 4: Using translators
+
+Previously we mentioned that there was no way for an end-user to customize the asset definitions, for example, changing the key of the asset. The recommended way to accomplish this is through the use of a translator class.
+
+This class provides end-users with a way to override the translation between a configuration file, and how that maps to concepts in Dagster.
+
+To start, we will define a translator method to map the table specification to a Dagster asset key associated, however, this approach can be mapped to many other attributes such as: dependencies, group name, metadata, and more. You can reference the implementation for other integrations for a full example. It is important to expose all use-cases when defining the parameters passed to the asset decorator, along with methods implemented on the translator.
```

Comment:
```suggestion
To start, we will define a translator method to map the table specification to a Dagster asset key. This approach, however, can be mapped to other attributes like dependencies, group names, metadata, and more. **Note**: When defining the parameters passed to the asset decorator, it's important to expose all use cases along with the methods implemented on the translator.
```

- Suggesting we break up that first sentence to make it easier to read
- About this guy:

  > You can reference the implementation for other integrations for a full example.

   Having this in the middle of this paragraph feels odd, and a little bit like we're telling the user to go find it themselves. Should they refer to the links included in the intro?
- Also suggesting we format the last sentence as `**Note**:` to make it stand out more. I'm not sure if my wording here is totally correct + what you intended, though